packages/flutter_tools/lib/src/ios/devices.dart

```diff
@@ -163,8 +164,8 @@ class IOSDevice extends Device {
     // TODO(devoncarew): Handle startPaused, debugPort.
     printTrace('Building ${app.name} for $id');
 
-    // Step 1: Install the precompiled application if necessary.
-    bool buildResult = await buildIOSXcodeProject(app, buildForDevice: true);
+    // Step 1: Install the precompiled/DBC application if necessary.
+    bool buildResult = await buildIOSXcodeProject(app, mode, buildForDevice: true);
```

Comment:
Flutter_tools will always issue a build before a startApp, so we won't get into a situation where the apk is built for the wrong mode than startApp would want.

Not for this PR, but I want to iteratively refactor the tool so that the Android and iOS workflows are substantially similar, and differences captured in the device subclasses. Right now we still have some instance of checks spread throughout the tool, to handle Android and iOS differently.